src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActions.tsx

```diff
@@ -96,40 +106,125 @@ export class ArtworkActions extends React.Component<
       })
   }
 
-  render() {
+  renderSaveButton() {
+    return <SaveButton artwork={this.props.artwork} render={Save(this.props)} />
+  }
+
+  renderViewInRoomButton() {
+    return (
+      <ContextConsumer>
+        {({ mediator }) => (
+          <UtilButton
+            name="viewInRoom"
+            onClick={() => this.openViewInRoom(mediator)}
+            label="View in room"
+          />
+        )}
+      </ContextConsumer>
+    )
+  }
+
+  renderShareButton() {
+    return (
+      <UtilButton
+        name="share"
+        onClick={this.toggleSharePanel.bind(this)}
+        label="Share"
+      />
+    )
+  }
+
+  renderDownloadButton() {
+    return (
+      <UtilButton
+        name="download"
+        href={this.getDownloadableImageUrl()}
+        label="Download"
+      />
+    )
+  }
+
+  renderEditButton() {
     const { artwork } = this.props
-    const downloadableImageUrl = this.getDownloadableImageUrl()
     const editUrl = `${sd.CMS_URL}/artworks/${artwork.id}/edit?current_partner_id=${artwork.partner.id}` // prettier-ignore
+
+    return <UtilButton name="edit" href={editUrl} label="Edit" />
+  }
+
+  renderGenomeButton() {
+    const { artwork } = this.props
     const genomeUrl = `${sd.GENOME_URL}/genome/artworks?artwork_ids=${artwork.id}` // prettier-ignore
 
+    return <UtilButton name="genome" href={genomeUrl} label="Genome" />
+  }
+
+  render() {
+    const { artwork } = this.props
+    const downloadableImageUrl = this.getDownloadableImageUrl()
+
+    const actionsToShow = [
```

Comment:
I did this to avoid some hairy conditionals. The short version is that in the `xs` case we need to show 3 buttons _max_ and hide the rest behind the "More" icon, but we don't know which buttons those are until we check _all_ of the conditions.